packages/ic-http-certification/src/http/http_response.rs

```diff
@@ -533,3 +533,40 @@ impl<'a> From<HttpResponse<'a>> for HttpUpdateResponse<'a> {
         }
     }
 }
+
+/// Asserts that two [HTTP responses](HttpResponse) are equal by comparing
+/// the status code, *sorted* headers, body, and upgrade flag of the two responses.
+///
+/// # Examples
+///
+/// ```
+/// use ic_http_certification::{HttpResponse, assert_response_eq};
+///
+/// let a = HttpResponse::builder()
+///     .with_status_code(200)
+///     .with_headers(vec![("Content-Type".into(), "text/plain".into())])
+///     .with_body(b"Hello, World!")
+///     .with_upgrade(false)
+///     .build();
+///
+/// let b = HttpResponse::builder()
+///     .with_status_code(200)
+///     .with_headers(vec![("Content-Type".into(), "text/plain".into())])
+///     .with_body(b"Hello, World!")
+///     .with_upgrade(false)
+///     .build();
+///
+/// assert_response_eq(a, b);
+/// ```
+pub fn assert_response_eq(a: HttpResponse, b: HttpResponse) {
```

Comment:
nit: not sure this should be a part of the main package, rather than some test-util-crate 🤔